src/EditorFeatures/CSharpTest/ValidateFormatString/ValidateFormatStringTests.cs

```diff
@@ -0,0 +1,625 @@
+        string.Format(""This {0} {1} {2} works[||]"", testParamsArray); 
+    }     
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task IFormatProviderandOnePlaceholder()
+        {
+            await TestMissingAsync(@" using System.Globalization; 
+class Program
+{
+    static void Main(string[] args)
+    {
+        CultureInfo culture = new System.Globalization.CultureInfo(""da - da"");
+        testStr = string.Format(culture, ""The current price [||]is {0:C2} per ounce"", 2.45);
+    }     
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task IFormatProviderandTwoPlaceholders()
+        {
+            await TestMissingAsync(@" using System.Globalization; 
+class Program
+{
+    static void Main(string[] args)
+    {
+        CultureInfo culture = new System.Globalization.CultureInfo(""da - da"");
+        testStr = string.Format(culture, ""The current price [||]is {0:C2} per {1} "", 2.45, ""ounce"");
+    }     
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task IFormatProviderandThreePlaceholders()
+        {
+            await TestMissingAsync(@" using System.Globalization; 
+class Program
+{
+    static void Main(string[] args)
+    {
+        CultureInfo culture = new System.Globalization.CultureInfo(""da - da"");
+        testStr = string.Format(culture, ""The current price [||]is {0} {1} {2} "", 2.45, ""per"", ""ounce"");
+    }     
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task IFormatProviderandFourPlaceholders()
+        {
+            await TestMissingAsync(@" using System.Globalization; 
+class Program
+{
+    static void Main(string[] args)
+    {
+        CultureInfo culture = new System.Globalization.CultureInfo(""da - da"");
+        testStr = string.Format(culture, ""The current price [||]is {0} {1} {2} {3} "", 2.45, ""per"", ""ounce"", ""today only"");
+    }     
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task IFormatProviderandParamsObjectArray()
+        {
+            await TestMissingAsync(@" using System.Globalization; 
+class Program
+{
+    static void Main(string[] args)
+    {
+        CultureInfo culture = new System.Globalization.CultureInfo(""da - da"");
+        object[] testParamsArray = { 1.25, ""2"", ""teststring""};
+        string.Format(culture, ""This {0} {1} {2} works[||]"", testParamsArray); 
+    }     
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
+        public async Task FormatStringWithComma()
+        {
+            await TestMissingAsync(@" class Program
+{
+    static void Main(string[] args)
+    {
+        string.Format(""{0,6}[||]"",34);
```

Comment:
This (and the next few tests) don't have a space after the comma between the parameters.  Unless that's important to the test (doesn't look like it), try to format the code in quotes as if it wasn't.

@jasonmalinowski How is your copious free time projection workspace for tests coming along?